src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/AppointmentService.kt

```diff
@@ -153,45 +147,90 @@ class AppointmentService(
   }
 
   @Transactional
-  fun updateAppointmentAddress(licenceId: Long, request: AddAddressRequest) {
-    val licenceEntity = licenceRepository
-      .findById(licenceId)
-      .orElseThrow { EntityNotFoundException("$licenceId") }
-
-    val previousAddress = licenceEntity.licenceAppointmentAddress?.toString()
-      ?: licenceEntity.appointmentAddress
+  fun addAppointmentAddress(licenceId: Long, request: AddAddressRequest) {
+    val licence = licenceRepository.findById(licenceId)
+      .orElseThrow { EntityNotFoundException("Licence $licenceId not found") }
+
+    val staff = getStaffUser()
+    val auditData = when (licence.licenceAppointmentAddress) {
+      null -> createAppointmentAddress(licence, request, staff)
+      else -> updateAppointmentAddress(licence, request, staff)
+    }
 
-    val username = SecurityContextHolder.getContext().authentication.name
+    licence.dateLastUpdated = LocalDateTime.now()
+    licence.updatedByUsername = staff?.username ?: SYSTEM_USER
+    licence.updatedBy = staff
 
-    val staffMember = staffRepository.findByUsernameIgnoreCase(username)
+    auditService.recordAuditEventInitialAppointmentUpdate(
+      licence,
+      auditData,
+      staff,
+    )
+  }
 
-    licenceEntity.licenceAppointmentAddress = AddressMapper.toEntity(request)
-    licenceEntity.appointmentAddress = request.toString()
-    licenceEntity.dateLastUpdated = LocalDateTime.now()
-    licenceEntity.updatedByUsername = staffMember?.username ?: SYSTEM_USER
-    licenceEntity.updatedBy = staffMember ?: licenceEntity.updatedBy
+  private fun createAppointmentAddress(licence: Licence, request: AddAddressRequest, staff: Staff?): Map<String, String> {
+    val address = addressMapper.toEntity(request)
+    val addressString = request.toString()
 
-    licenceRepository.saveAndFlush(licenceEntity)
+    licence.appointmentAddress = addressString
+    licence.licenceAppointmentAddress = address
 
-    if (request.isPreferredAddress && staffMember != null) {
-      staffMember.savedAppointmentAddresses.add(AddressMapper.toEntity(request))
-      staffRepository.saveAndFlush(staffMember)
```

Comment:
This was the error 
`AddressMapper.toEntity(request)`
it was not connected to he one licenceAppointmentAddress therefore it was creating two rows in the DB